src/lib/blockchain_snark/blockchain_snark_state.ml

```diff
@@ -374,7 +374,7 @@ let rule ~proof_level ~constraint_constants transaction_snark self :
         ; public_output = ()
         ; auxiliary_output = ()
         } )
-  ; uses_lookup = false
+  ; feature_flags = Pickles_types.Plonk_types.Features.none_bool
```

Comment:
For inductive proofs do the previous proofs need to have the same feature flags as:
* the wrapping proof
* each other?

Reply:
They don't, no. We pass in a 'tag' that describes each of the previous proofs' features (among other things, like the public input type), which can be distinct from the tags for the rule. The features for the circuits are determined from the union of all of the features used by all rules [here](https://github.com/MinaProtocol/mina/blob/01ac51b8126cfcf44e458547746794793dd67c62/src/lib/pickles/pickles.ml#L508), which are then embedded in those tags.

Reply:
Security question: I guess a verifier cannot be fooled by "feature flags malleability" (someone mutating them and sending them off with a proof that was generated with different feature flags) because the flags end up implicitly baked into the verifier index?